streaming/base/dataset.py

```diff
@@ -241,39 +245,40 @@ def __init__(self,
         self.shards = []
         stream_per_shard = []
-        self.shard_offset_per_stream = np.zeros(len(streams), np.int64)
-        self.shards_per_stream = np.zeros(len(streams), np.int64)
-        self.sample_offset_per_stream = np.zeros(len(streams), np.int64)
-        self.samples_per_stream = np.zeros(len(streams), np.int64)
-        for idx, stream in enumerate(streams):
+        self.shard_offset_per_stream = np.zeros(self.num_streams, np.int64)
+        self.shards_per_stream = np.zeros(self.num_streams, np.int64)
+        self.sample_offset_per_stream = np.zeros(self.num_streams, np.int64)
+        self.samples_per_stream = np.zeros(self.num_streams, np.int64)
+        for stream_id, stream in enumerate(self.streams):
             stream_shards = stream.get_shards(world)
             samples = sum(map(len, stream_shards))
-            stream_per_shard += [idx] * len(stream_shards)
-            self.shard_offset_per_stream[idx] = len(self.shards)
-            self.shards_per_stream[idx] = len(stream_shards)
-            self.sample_offset_per_stream[idx] = self.num_samples
-            self.samples_per_stream[idx] = samples
+            stream_per_shard += [stream_id] * len(stream_shards)
+            self.shard_offset_per_stream[stream_id] = len(self.shards)
+            self.shards_per_stream[stream_id] = len(stream_shards)
+            self.sample_offset_per_stream[stream_id] = self.num_samples
+            self.samples_per_stream[stream_id] = samples
             self.shards += stream_shards
             self.num_samples += samples
         self.stream_per_shard = np.array(stream_per_shard, np.int64)
+        self.num_shards = len(self.shards)
 
         # Build the Index (for partitioning and mapping samples to shards).
-        self.shard_sizes = np.array([x.samples for x in self.shards])
-        self.index = Index(self.shard_sizes)
+        self.samples_per_shard = np.array([x.samples for x in self.shards])
+        self.index = Index(self.samples_per_shard)
 
         # Now that we have the true size of each sub-dataset, derive the proportions/repeats/picks.
         if is_proportional:
             # Relative.
             if not samples_per_epoch:
                 samples_per_epoch = self.index.total_samples
-            self.proportion_per_stream = np.array([stream.proportion for stream in streams],
+            self.proportion_per_stream = np.array([stream.proportion for stream in self.streams],
                                                   np.float64)
             self.proportion_per_stream /= self.proportion_per_stream.sum()
             self.pick_per_stream = (samples_per_epoch * self.proportion_per_stream).astype(
                 np.int64)
             short = samples_per_epoch - self.pick_per_stream.sum()
             rng = np.random.default_rng(shuffle_seed)
-            indices = rng.choice(len(self.pick_per_stream), short, False)
+            indices = rng.choice(self.num_streams, short, False)
```

Comment:
Is changing `len(self.pick_per_stream)` to `self.num_streams` is correct ?

Reply:
yes, there is one value per stream in pick_per_stream